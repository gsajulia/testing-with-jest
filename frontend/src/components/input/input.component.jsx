import './input.styles.css';

export const Input = ({ searchValue, handleChange }) => {
    return (
        <input
            onChange={handleChange}
            value={searchValue}
            className="text-input"
            type="search"
            placeholder="Type your search" />
    )
}