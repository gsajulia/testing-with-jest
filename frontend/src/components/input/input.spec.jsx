import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Input } from './input.component';

describe('<Input />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<Input handleChange={fn} searchValue={'testando'} />);

    const input = screen.getByPlaceholderText(/type your search/i);
    expect(input.value).toBe('testando');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<Input handleChange={fn} />);

    const input = screen.getByPlaceholderText(/type your search/i);
    const value = 'o valor';

    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });
});