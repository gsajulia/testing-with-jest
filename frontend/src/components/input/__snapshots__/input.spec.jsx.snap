// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<div>
  <input
    class="text-input"
    placeholder="Type your search"
    type="search"
    value=""
  />
</div>
`;
